core/services/headreporter/telemetry_reporter.go

```diff
@@ -0,0 +1,53 @@
+package headreporter
+
+import (
+	"context"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/legacyevm"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/services/telemetry"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+	"math/big"
+)
+
+type (
+	telemetryReporter struct {
+		logger    logger.Logger
+		endpoints map[*big.Int]commontypes.MonitoringEndpoint
+	}
+)
+
+func NewTelemetryReporter(chainContainer legacyevm.LegacyChainContainer, lggr logger.Logger, monitoringEndpointGen telemetry.MonitoringEndpointGenerator) *telemetryReporter {
+	endpoints := make(map[*big.Int]commontypes.MonitoringEndpoint)
+	for _, chain := range chainContainer.Slice() {
+		endpoints[chain.ID()] = monitoringEndpointGen.GenMonitoringEndpoint("EVM", chain.ID().String(), "", synchronization.HeadReport)
+	}
+	return &telemetryReporter{
+		logger:    lggr,
+		endpoints: endpoints,
+	}
+}
+
+func (t *telemetryReporter) reportOnHead(ctx context.Context, head *evmtypes.Head) {
+	monitoringEndpoint := t.endpoints[head.EVMChainID.ToInt()]
+	request := &telem.HeadReportRequest{
+		ChainId:     head.EVMChainID.String(),
+		Timestamp:   uint64(head.Timestamp.UTC().Unix()),
+		BlockNumber: uint64(head.Number),
+		BlockHash:   head.Hash.Hex(),
+		Finalized:   head.IsFinalized,
+	}
```

Comment:
You need this to be something like:
```go
type Block struct {
  Timestamp uint64
  Number    uint64
  Hash      string
}
type HeadReportRequest {
  ChainId   uint64
  Latest    Block
  Finalized Block
}
```

Use `head.LatestFinalizedHead()` to fetch finalized for a given latest head.

Reply:
i.e. we want block info for both `latest` and `finalized` (at that latest point in time)

Reply:
done